cmd/podman/compose.go

```diff
@@ -184,12 +188,18 @@ func composeDockerHost() (string, error) {
 		if err != nil {
 			return "", fmt.Errorf("inspecting machine: %w", err)
 		}
-		if info.ConnectionInfo.PodmanSocket == nil {
-			return "", errors.New("socket of machine is not set")
-		}
 		if info.State != machine.Running {
 			return "", fmt.Errorf("machine %s is not running but in state %s", item.Name, info.State)
 		}
+		if item.VMType == "wsl" {
```

Comment:
```
if machineProvider.VMType() == machine.WSLVirt
```

might be stronger type comparison ... though I readily admit I am not sure if that causes cyclical dependency issues or not.